packages/@aws-cdk/aws-ec2-alpha/lib/ipam.ts

```diff
@@ -343,6 +345,11 @@ class IpamPool extends Resource implements IIpamPool {
       throw new Error('awsService is required when addressFamily is set to ipv6');
     }
 
+    //Add tags to the IPAM Pool if name is provided
+    if (props.ipamPoolName) {
+      Tags.of(this).add(NAME_TAG, props.ipamPoolName);
```

Comment:
What if users want to add a custom tag that also called this name?

Reply:
This is added according to the resource name provided, even if they want to add tag i wonder they would like it to rename it to some other value than the actual name of the resource.